services/worker-manager/src/data.js

```diff
@@ -123,54 +123,59 @@ WorkerPool.expire = async function(monitor) {
 WorkerPool.prototype.reportError = async function({kind, title, description, extra={}}) {
   const errorId = slugid.v4();
 
-  if (this.emailOnError) {
-    let extraInfo = '';
-    if (Object.keys(extra).length) {
-      extraInfo = `
-It includes the extra information:
-
-\`\`\`
-${yaml.safeDump(extra)}
-\`\`\`
-      `.trim();
-    }
-    await this.notify.email({
-      address: this.owner,
-      subject: `Taskcluster Worker Manager Error: ${title}`,
-      content: `
-Worker Manager has encountered an error while trying to provision the worker pool ${this.workerPoolId}:
-
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-${description}
+  try {
+    if (this.emailOnError) {
+      let extraInfo = '';
+      if (Object.keys(extra).length) {
+        extraInfo = `
+            It includes the extra information:
+            
+            \`\`\`
+            ${yaml.safeDump(extra)}
+            \`\`\`
+                  `.trim();
+      }
 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      await this.notify.email({
+        address: this.owner,
+        subject: `Taskcluster Worker Manager Error: ${title}`,
+        content: `
+            Worker Manager has encountered an error while trying to provision the worker pool ${this.workerPoolId}:
+            
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            
+            ${description}
+            
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            
+            ErrorId: ${errorId}
+            
+            ${extraInfo}
+      `.trim(),
+      });
+    }
 
-ErrorId: ${errorId}
+    await this.monitor.log.workerError({
```

Comment:
This should probably be in the `finally` block, too.

Reply:
I want only one function call to be in the `finally`, the most important one (which, I judge, is the `workerPoolError` thing because it's displayed in the UI. If there are two functions in the `finally`, and the first errors out, the second won't get executed.

Reply:
Another way to go about this can be orchestrating a series of try-catch blocks, one per function call, if we'd rather consider all three to be equally important

Reply:
I was thinking of just running all three things in parallel - they're all important, after all.

Reply:
I'm going to merge this as is for now. I'd like to use `Promise.allSettled` here to make sure the first two functions work independently in parallel, and then the third function is executed in `.then()` or `.finally()` of that. I need Node 12 for all this awesomeness tho